### PR TITLE
Squashable updates to add-enrollment-reports branch

### PR DESCRIPTION
**Pull request by fghaas**
* Fix typos and flake8 issues
* Make `MAIL_FROM` default to `SMTP_USERNAME`
* Make `MAIL_TO` a list
* Remove `run_enrollment_reports.sh` (it serves no purpose: the Postfix configuration can't be baked into the Docker image, and the remaining `ansible-playbook` invocation doesn't need a script)
* Rename `config.yml` (which isn't a YAML file) to `my.cnf`
* Clean up the Dockerfile and drop anything that can't be baked in.

What still needs doing after this is merged is properly including the Ansible inventory and `my.cnf`, via volumes.

### CODE REVIEW

**Comment by fghaas** [comment]
@foadlind if you could lend me your eyes for an extra look at these, I'd appreciate that.

**Comment by foadlind** [comment]
It looks good to me. I have one suggestion: Unless `my.cnf ` is specifically needed somewhere, you could skip it altogether and just set these env vars in docker compose when defining the service:
```
- MYSQL_HOST={{ MYSQL_HOST }}
- MYSQL_PORT={{ MYSQL_PORT }}
- MYSQL_ROOT_USERNAME={{ MYSQL_ROOT_USERNAME }}
- MYSQL_ROOT_PASSWORD={{ MYSQL_ROOT_PASSWORD }}
```
Then they are available inside the container as env vars and can be fed to `default-mysql-client`.

**Comment by fghaas** [comment]
Hmmm, checking [the MySQL 5.7 docs](https://dev.mysql.com/doc/refman/5.7/en/environment-variables.html) and [the MariaDB docs](https://mariadb.com/kb/en/mariadb-environment-variables/) it would appear that the environment variables to set are `MYSQL_HOST`, `MYSQL_PWD`, `MYSQL_TCP_PORT` and `USER`; would you agree with that reading?

Also, I am not 100% sure that these environment variables get passed into the execution environment of the Ansible playbook. They should be, but we should definitely test to make sure.

File: tutorenrollmentreports/templates/enrollmentreports/build/enrollmentreports/Dockerfile [hunk 1]
```diff
@@ -2,20 +2,12 @@ FROM python:3.8
...
-RUN git clone https://github.com/hastexo/ansible-role-enrollmentreports.git /etc/ansible/roles/enrollment_report
+RUN git -C /etc/ansible/roles clone https://github.com/hastexo/ansible-role-enrollmentreports.git enrollmentreports
 RUN echo "postfix postfix/main_mailer_type string Internet site" > postfix_install.txt
-RUN echo "postfix postfix/mailname string mail.example.com" >> postfix_install.txt
 RUN debconf-set-selections postfix_install.txt
 RUN DEBIAN_FRONTEND=noninteractive apt-get install -q -y postfix
 COPY enrollment-report.yml .
-COPY run_enrollment_reports.sh .
-COPY inventory/config.yml /etc/ansible/group_vars/all
-RUN  useradd mysql
```
Comment (NamrataSitlani) [inline tutorenrollmentreports/templates/enrollmentreports/build/enrollmentreports/Dockerfile:20]: If I run the playbook, without adding `mysql` user explicitly I see the following Ansible file module error :
` chown failed: failed to look up user mysql`

File: tutorenrollmentreports/plugin.py [hunk 2]
```diff
@@ -34,14 +35,14 @@
...
-@local_command_group.command(help="Run the enrollment reports scrio")
+@local_command_group.command(help="Run the enrollment reports script")
 @click.pass_obj
 def enrollmentreports(context):
     config = tutor_config.load(context.root)
     job_runner = context.job_runner(config)
     job_runner.run_job(
         service="enrollmentreports",
-        command=f"bash -e run_enrollment_reports.sh"
+        command="ansible-playbook enrollment-report.yml"
```
Comment (NamrataSitlani) [inline tutorenrollmentreports/plugin.py:45]: I am unable to understand this bit, if we don't use the bash script to start the `postfix` service and update the configurations. How are we going to take care of postfix service inside the container?

Comment (fghaas) [reply tutorenrollmentreports/plugin.py:45]: Whichever way we do, it definitely won't be in such a way that you bake a script into the Docker image that contains the Postfix configuration.

**Comment by foadlind** [comment]
> Hmmm, checking [the MySQL 5.7 docs](https://dev.mysql.com/doc/refman/5.7/en/environment-variables.html) and [the MariaDB docs](https://mariadb.com/kb/en/mariadb-environment-variables/) it would appear that the environment variables to set are MYSQL_HOST, MYSQL_PWD, MYSQL_TCP_PORT and USER; would you agree with that reading?

Hmm ok I didn't know that. In the official MySQL image they are named differently: https://hub.docker.com/_/mysql

**Comment by foadlind** [comment]
OK I see my error. The ones I mentioned are for the MySQL image. The ones @fghaas mentioned are the ones MySQL client will look for.

**Comment by fghaas** [comment]
> In the official MySQL image they are named differently

Those are the variables it understands for the MySQL *server* (`mysqld`); we are looking to configure the MySQL *client* (`mysql`).

File: tutorenrollmentreports/templates/enrollmentreports/build/enrollmentreports/Dockerfile (hunk 1 above)
Comment (fghaas) [reply tutorenrollmentreports/templates/enrollmentreports/build/enrollmentreports/Dockerfile:20]: Yeah on second thought, it doesn't really make a ton of sense to base this image on `python:3.8`. See 8b836e4abaf0aa1b1265ade0b2d97ffbcfa580bc.

Comment (NamrataSitlani) [reply tutorenrollmentreports/templates/enrollmentreports/build/enrollmentreports/Dockerfile:20]: Acknowledged.

**Comment by fghaas** [comment]
> Yeah on second thought, it doesn't really make a ton of sense to base this image on python:3.8. See 8b836e4abaf0aa1b1265ade0b2d97ffbcfa580bc.

On third thought, neither does it make sense to base things on the `mysql` image because that just starts `mysqld` on container startup. Update is in 52c78436896ffb14383851a28dcb5079088bc7ee. 

The problem that we previously relied on the presence of the `mysql` user should be fixed with https://github.com/hastexo/ansible-role-enrollmentreports/pull/2.

**Comment by NamrataSitlani** [comment]
@foadlind If you have reviewed this PR, should I go ahead and merge this? Thanks!

**Comment by foadlind** [comment]
@NamrataSitlani yes you can merge it.

**Pull request merged**